File: 3860-resulting-string-after-adjacent-removals/3860-resulting-string-after-adjacent-removals.go
func resultingString(s string) string {
    stk := make([]byte, 0, len(s))
    
	help := func(a, b byte) bool {
		v := (int(a) - int(b) + 26) % 26 
		return v == 25 || v == 1
	}
    
	for i := 0; i < len(s); i++ {
		c := s[i]
		if len(stk) != 0 && help(stk[len(stk) - 1], c) {
			stk = stk[ : len(stk) - 1]
		} else {
			stk = append(stk, c)
		}
	}
	return string(stk)
}

File: 3860-resulting-string-after-adjacent-removals/3860-resulting-string-after-adjacent-removals.cpp
class Solution {
public:
    string resultingString(string s) {
        string stk;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            if (!stk.empty() and (fabs(stk.back() - s[i]) == 1 or fabs(stk.back() - s[i]) == 25))
                stk.pop_back();
            else
                stk.push_back(s[i]);
        }
        
        /*string t = "";
        while (!stk.empty()) {
            t = stk.back() + t;
            stk.pop_back();
        }*/
        
        return stk;
    }
};